### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by FearsomeRover**
Added saving auth jwt to cookies, and an example request using that auth to get the users own submissions.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -19,5 +19,6 @@
   },
   "[prisma]": {
     "editor.defaultFormatter": "Prisma.prisma"
-  }
+  },
+  "prettier.configPath": "packages/backend/.prettierrc"
```
Comment (Tschonti) [inline .vscode/settings.json]: That might not be the best solution, we might want different settings for the frontend (e. g., in react projects , we usually  use double quotes in JSX (like name="Samu"), but single quote everywhere else). But we can figure this out some other time

File: packages/client/src/app/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input type='email' name='email' placeholder='Email' required />
+      <input type='password' name='password' placeholder='Password' required />
+      <button type='submit'>Login</button>
+    </form>
+  )
+}
```
Comment (Tschonti) [inline packages/client/src/app/login/page.tsx:34]: what's the purpose of this page? xd afaik we don't want username-pwd auth

File: packages/client/src/network/getSubmissions.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const cookieStore = cookies()
+  console.log(cookieStore.get('jwt'))
+  console.log(cookieStore.getAll())
+  try {
+    const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/submissions/me`, {
+      cache: 'no-store',
+      headers: {
+        Authorization: `Bearer ${cookieStore.get('jwt')?.value}`,
+      },
+    })
```
Comment (Tschonti) [inline packages/client/src/network/getSubmissions.ts]: we probably have to create a wrapper function for the fetch that will inject the token. The next fetch doesn't seem to have interceptors like axios :(

**Review by Tschonti — APPROVED** [review]
thanks, it gets the job done!

If you could remove the unnecessary console logs and comments, that'd be nice

File: packages/client/src/app/login/page.tsx (hunk 2 above)
Comment (FearsomeRover) [reply packages/client/src/app/login/page.tsx:34]: I have no idea, why that is there...

File: .vscode/settings.json (hunk 1 above)
Comment (FearsomeRover) [reply .vscode/settings.json]: Prettier was looking red without a config, and this seemed the easy solution, but i agree, that we should do something about this.

**Comment by FearsomeRover** [comment]
I added some axios config, it works, and looks clean when used in functions.
 `const response = await api.get('/submissions/me')`
It also handles adding the base url.

**Comment by Tschonti** [comment]
Nice, but this way we lose the cahcing feature of the next-fetch. But we probably won't use that too much, and in those cases, we can still just add the header manually.

**Review by Tschonti — APPROVED** [review]

**Pull request merged**